Restructure obsv overview (#13119)
kyma-project/kyma: docs/01-overview/main-areas/observability/obsv-01-monitoring-in-kyma.md

---
title: Monitoring in Kyma
---

> [Prometheus](https://prometheus.io/) is the open source monitoring and alerting toolkit that provides the telemetry data. This data is consumed by different addons, including [Grafana](https://grafana.com/) for analytics and monitoring, and [Alertmanager](https://prometheus.io/docs/alerting/alertmanager/) for handling alerts.

Reviewer: ```suggestion
[Prometheus](https://prometheus.io/) is the open source monitoring and alerting toolkit that collects and stores metrics data. This data is consumed by different addons, including [Grafana](https://grafana.com/) for analytics and monitoring, and [Alertmanager](https://prometheus.io/docs/alerting/alertmanager/) for handling alerts.
```

Monitoring in Kyma is configured to collect all metrics relevant for observing the in-cluster [Istio](https://istio.io/latest/docs/concepts/observability/) Service Mesh. For diagrams of the default setup and the monitoring flow with Istio, see [Istio Monitoring Architecture](../../../05-technical-reference/00-architecture/obsv-01-architecture-monitoring-istio.md). Learn how to [enable Grafana visualization](../../../04-operation-guides/operations/obsv-03-enable-grafana-for-istio.md) and [enable mTLS for custom metrics](../../../04-operation-guides/operations/obsv-04-enable-mtls-istio). 

## Limitations

> By default, Prometheus stores up to **15 GB** of data for a maximum period of **30 days**. If the default size or time is exceeded, the oldest records are removed first.

Reviewer: Only valid in the production profile. In the "no profile" it has a retention of 2GB for 1d..

Maybe we can incorporate the "production" term in the title?

Author: ```suggestion
By default, in the production profile, Prometheus stores up to **15 GB** of data for a maximum period of **30 days**. If the default size or time is exceeded, the oldest records are removed first.
```

> The configured memory limits of the Prometheus and Prometheus-Istio instances limit the number of time series samples that can be ingested. 

Reviewer: The limits limit the timeseries... sounds bad

Author: ```suggestion
The configured memory limits of the Prometheus and Prometheus-Istio instances define the number of time series samples that can be ingested. 
```

> The default resource configuration of the monitoring component in the production profile is sufficient to serve **800K time series in the Prometheus Pod**, and **400K time series in the Prometheus-Istio Pod**. The samples are deleted after 30 days or when reaching the storage limit of 30 GB. The evaluation profile has lower limits. For more information about profiles, see [Install Kyma: Choose resource consumption](../../../04-operation-guides/operations/02-install-kyma.md#choose-resource-consumption).

Reviewer: Here the production profile gets mentioned..

---

> title: Monitoring in Kyma

Author: Good point!

---

[Prometheus](https://prometheus.io/) is the open source monitoring and alerting toolkit that collects and stores metrics data. This data is consumed by different addons, including [Grafana](https://grafana.com/) for analytics and monitoring, and [Alertmanager](https://prometheus.io/docs/alerting/alertmanager/) for handling alerts.

> Monitoring in Kyma is configured to collect all metrics relevant for observing the in-cluster [Istio](https://istio.io/latest/docs/concepts/observability/) Service Mesh. For diagrams of the default setup and the monitoring flow with Istio, see [Istio Monitoring Architecture](../../../05-technical-reference/00-architecture/obsv-01-architecture-monitoring-istio.md). Learn how to [enable Grafana visualization](../../../04-operation-guides/operations/obsv-03-enable-grafana-for-istio.md) and [enable mTLS for custom metrics](../../../04-operation-guides/operations/obsv-04-enable-mtls-istio). 

Author: will fix it, thanks for noticing!

## Limitations

> In the production profile, Prometheus stores up to **15 GB** of data for a maximum period of **30 days**. If the default size or time is exceeded, the oldest records are removed first.

Author: ```suggestion
In the [production profile](../../../04-operation-guides/operations/02-install-kyma.md##choose-resource-consumption), Prometheus stores up to **15 GB** of data for a maximum period of **30 days**. If the default size or time is exceeded, the oldest records are removed first. The evaluation profile has lower limits.
```

> By default, in the production profile, Prometheus stores up to **15 GB** of data for a maximum period of **30 days**. If the default size or time is exceeded, the oldest records are removed first.

Author: ```suggestion
In the production profile, Prometheus stores up to **15 GB** of data for a maximum period of **30 days**. If the default size or time is exceeded, the oldest records are removed first.
```

The configured memory limits of the Prometheus and Prometheus-Istio instances define the number of time series samples that can be ingested. 

> The default resource configuration of the monitoring component in the production profile is sufficient to serve **800K time series in the Prometheus Pod**, and **400K time series in the Prometheus-Istio Pod**. The samples are deleted after 30 days or when reaching the storage limit of 30 GB. 

Reviewer: ```suggestion
The default resource configuration of the monitoring component in the production profile is sufficient to serve **800K time series in the Prometheus Pod**, and **400K time series in the Prometheus-Istio Pod**. The samples are deleted after 30 days or when reaching the storage limit of 15 GB. 
```

> The evaluation profile has lower limits. For more information about profiles, see [Install Kyma: Choose resource consumption](../../../04-operation-guides/operations/02-install-kyma.md#choose-resource-consumption).

Author: Yes I like your idea where to put the link much more! I'm not sure about the bullets, as I don't want to spell out the eval limits (lower maintenance effort), and it might look weird to have one very detailed bullet and one very vague :D

> The evaluation profile has lower limits. For more information about profiles, see [Install Kyma: Choose resource consumption](../../../04-operation-guides/operations/02-install-kyma.md#choose-resource-consumption).

Author: ```suggestion
```

The amount of generated time series in a Kyma cluster depends on the following factors:

* Number of Pods in the cluster
* Number of Nodes in the cluster
* Amount of exported (custom) metrics
* Label cardinality of metrics
* Number of buckets for histogram metrics
* Frequency of Pod recreation

> * Topology of the Istio service mesh

Reviewer 2: ```suggestion
* Topology of the Istio Service Mesh
```

Reviewer 2: ^ not sure about this one, though 🤔
